README.md

```diff
@@ -0,0 +1,215 @@
+
+Yes, this was confusing. That's why in A3, the `name` option no longer exists, and **the name of the module and the `type` property in the database are always the same.** Also, by convention, module names in A3 are singular.
+
+What does this mean for us when we upgrade the content? It means that most of the time, **in our A3 source code we need to change the module name to singular and remove its `name` option, but we don't need to change the `type` in the database,** because the new module's name will likely match it.
+
+However, if you *do* want to remap the `type` in the database, perhaps because you're choosing a different name for your module in A3, you can do it like this:
+
+```javascript
+module.exports = {
+  mapDocTypes: {
+    'old-name': 'new-name',
+    ...
+  }
+}
+```
+
+🎩 **`old-name` must match the `name` option from the old A2 module,** not the module name from A2. `new-name` must match the module name in your new A3 project, which will also be used for the `type` property.
+
+#### Providing a transformation function
+
+Usually just remapping the name is enough. However if you have designed a new schema for a doc type in your A3 project and you want to migrate the A2 data to that new format, you can supply a transformation function:
+
+```javascript
+module.exports = {
+  mapDocTypes: {
+    // Async functions are allowed and will be awaited
+    'old-name': async (doc) => {
+      return {
+        ...doc,
+        // We've decided to change the name
+        type: 'new-name',
+        // We've changed to a single string field for the address
+        address: `${doc.street} ${doc.city}, ${doc.state}`
+      };
+    }
+  }
+}
+```
+
+🎩 **Your function **must** return a doc unless you want the document to be **removed** in the upgrade.** It's OK to modify the original doc but you must return the modified doc if you want it to be kept.
+
+### Transforming widget types
+
+Just like doc types, in A2 widget `type` properties were set via the `name` option of the widget module in question. This was *set automatically* in most cases based on the module name, **with the `-widgets` part removed.** 
+
+In A3 it is similar, but the `name` option is no longer supported, and the suffix removed from the module name is `-widget`.
+
+For most widgets this means you should not need to remap the type names. You can just rename the module from `-widgets` to `-widget`.
+
+If you choose though, you can rename the types like this:
+
+```javascript
+module.exports = {
+  mapWidgetTypes: {
+    'old-name': 'new-name',
+    ...
+  }
+}
+```
+
+#### Transformation functions and pieces-widgets
+
+Keep in mind that **A3 does not have a direct equivalent to `apostrophe-pieces-widgets`. So when you upgrade your code, you'll likely change those widget modules to extend `@apostrophecms/widget-type` and just use a `relationship` schema field to select pieces. The catch is moving the data. A migration function can help you do that:
```

Comment:
Isn't part of the direction for pieces widgets to first create (or plan out) a custom pieces widget replacement? Depending on your use that new widget may need all the original fields or only one.

Reply:
All of these use cases are discussed below in the way you're suggesting, so I'm not sure what else to do here.